Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/CartesianCKCAlgorithm.H

```diff
@@ -135,6 +135,23 @@ struct CartesianCKCAlgorithm {
         return inv_dx*( F(i,j,k,ncomp) - F(i-1,j,k,ncomp) );
     };
 
+#ifdef WARPX_MAG_LLG
+    /**
+     * Perform derivative along x on a nodal grid, from a cell-centered field `F`*/
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real DownwardMagDx (
+        amrex::Array4<amrex::Real> const& F,
+        amrex::Array4<amrex::Real> const& M,
+        amrex::Real const * const coefs_x, int const n_coefs_x,
+        int const i, int const j, int const k, amrex::Real const mu, int const ncomp=0, int const ncompM = 0 ) {
+
+        amrex::Real const inv_dx = coefs_x[0];
+        amrex::Real const inv_mu = 1.0/mu;
```

Comment:
there is no "mu" if we use B and M. Only "PhysConst::mu0" is available

Reply:
That is right. 

Since no user-defined variables are used in the CartesianCKCAlgorithm.H by convention,  it is convenient to pass the mu as an input variable. 

An alternative way is to define a DownwardMagDy only for the M field, yet it will not explicitly show the calculation of the H field. So I did not choose this way.

Reply:
Oh I didn't spot this is CKC. No H functions should use CKC. Just delete all these and set a flag in you DownwardMagDz "Magnetic materials are not included in CKC algorithm"